sources/api/thar-be-settings/src/main.rs

```diff
@@ -179,13 +179,20 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
                 "Requesting affected services for settings: {:?}",
                 &changed_settings
             );
-            let services =
-                service::get_affected_services(&args.socket_path, Some(changed_settings)).await?;
-            trace!("Found services: {:?}", services);
-            if services.is_empty() {
+
+            let service_settings_map =
+                service::get_affected_service_setting_map(&args.socket_path, changed_settings)
```

Comment:
I think my confusion mainly comes from the different ways we're storing settings and services, since some are duplicated, and what "services" actually means in different places.  Before, services was a `model::Services` object, but now we also have maps with just their names so we can find related settings.

Let me throw out an idea, because it might help make my confusion clearer, and at least give us something to start with.  What if we get rid of the change on this line, and change `get_affected_services` to return a new type that has all of that information together.  The new type, perhaps just called `Services`, would effectively be a map where the keys are service names, just like `service_settings_map` and `services` are now, and the values would be another new type (`Service`?) with one field for the relevant setting names and one field for the `Service` objects used in the model.  Then we'd have everything we need, and we can just pass that around instead of needing two different representations of services.  It'd be easy to convert for places where we need a `model::Services` or just the map of services to settings, or we could change those places to understand the new type instead of converting at all.

I think this could simplify quite a bit because we wouldn't have as many conversions between different representations, there'd be fewer maps, fewer naming issues, etc.  I can work on a prototype if it'd help clarify what I mean.

Reply:
Ok, I did [a prototype](https://gist.github.com/tjkirch/dbfc3be1b20ec4c6c52bf6d1e9d33139) of the `Services` thing I mentioned.  (prototype because it's untested, commits aren't split nicely, etc.)  @webern and @sanu11, what do you think, does this make the flow clearer?

Reply:
Thanks for putting it up @tjkirch  : ) Will go through it

Reply:
- `pub fn get_config_file_names(services: &Services)` could be a member function in `impl Services`.
- creating a member function for `services.0.is_empty()` improves readability slightly.

In general in both the original and @sanu11 's PR, I found the propagation of an `Option` to reflect `all services` very confusing. I suggested the change where @sanu11 introduced:
- `service::get_affected_services`
- `service::get_all_services`

I think hiding this implementation detail helps a bit instead of having mysterious options in `async fn run()`.

Reply:
I think there are too many functions and it's too hard to trace.  If the Option is what's confusing, we could make our own enum..? `get_services(Services::All)`, `get_services(Services::AffectedBySettings(list)`